src/raftstore/store/worker/region.rs

```diff
@@ -316,7 +482,32 @@ impl Runnable<Task> for Runner {
                 region_id,
                 start_key,
                 end_key,
-            } => self.ctx.handle_destroy(region_id, start_key, end_key),
+            } => {
+                if self.ctx.get_delay_secs() > 0
+                    && !self.ctx.destroy_overlap_ranges(&start_key, &end_key)
```

Comment:
I can't understand the logic here.

Reply:
If there are overlaps, delete overlaps in normal way, and register the new range to pending task.